tarodb/assets_common.go

```diff
@@ -334,3 +337,81 @@ func fetchGenesis(ctx context.Context, q FetchGenesisStore,
 		Type:         asset.Type(gen.AssetType),
 	}, nil
 }
+
+type GroupStore interface {
+	FetchGenesisStore
+
+	// AllGroups fetches all asset groups on disk.
+	AllGroups(context.Context) ([]sqlc.KeyGroupInfoView, error)
+}
+
+// FetchAllGroups fetches all asset groups on disk.
+func fetchAllGroups(ctx context.Context, q GroupStore) ([]*asset.AssetGroup, error) {
+
+	var (
+		dbGroupInfo []sqlc.KeyGroupInfoView
+		groups      []*asset.AssetGroup
+		err         error
+	)
+
+	dbGroupInfo, err = q.AllGroups(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch asset groups: %w", err)
+	}
+
+	// Look up matching genesis info for each group. Each group
+	// should have genesis info populated.
+	for _, g := range dbGroupInfo {
+		groupGenesis, err := fetchGenesis(
+			ctx, q, g.GenAssetID,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to fetch group "+
+				"genesis info: %w", err)
+		}
+
+		groupKey, err := parseGroupInfo(g)
+		if err != nil {
+			return nil, err
+		}
+
+		groups = append(groups, &asset.AssetGroup{
```

Comment:
I think we should pre-allocate the `groups` slice after we queried `q.AllGroups` and then do an index based insert here.